test/integration/integration.cc

```diff
@@ -300,16 +300,44 @@ void BaseIntegrationTest::createEnvoy() {
       ports.push_back(upstream->localAddress()->ip()->port());
     }
   }
+
+  if (use_lds_) {
+    ENVOY_LOG_MISC(debug, "Setting up file-based LDS");
+    // Before finalization, set up a real lds path, replacing the default /dev/null
+    std::string lds_path = TestEnvironment::temporaryPath(TestUtility::uniqueFilename());
+    config_helper_.addConfigModifier(
+        [lds_path](envoy::config::bootstrap::v2::Bootstrap& bootstrap) -> void {
+          bootstrap.mutable_dynamic_resources()->mutable_lds_config()->set_path(lds_path);
+        });
+  }
+
   // Note that finalize assumes that every fake_upstream_ must correspond to a bootstrap config
   // static entry. So, if you want to manually create a fake upstream without specifying it in the
   // config, you will need to do so *after* initialize() (which calls this function) is done.
   config_helper_.finalize(ports);
 
+  envoy::config::bootstrap::v2::Bootstrap bootstrap = config_helper_.bootstrap();
+  if (use_lds_) {
+    // After the config has been finalized, write the final listener config to the lds file.
+    const std::string lds_path = config_helper_.bootstrap().dynamic_resources().lds_config().path();
+    envoy::api::v2::DiscoveryResponse lds;
+    lds.set_version_info("0");
+    for (auto& listener : config_helper_.bootstrap().static_resources().listeners()) {
+      ProtobufWkt::Any* resource = lds.add_resources();
+      resource->PackFrom(listener);
+    }
+    TestEnvironment::writeStringToFileForTest(lds_path, MessageUtil::getJsonStringFromMessage(lds),
+                                              true);
+
+    // Now that the listeners have been written to the lds file, remove them from static resources
+    // or they will not be reloadable.
```

Comment:
I'm curious how does static_resources() get reloaded?

Reply:
Sorry, what I mean is after clearing static resources, how does a modified listener config in file get reloaded?

Reply:
Ah, gotcha.
So by sticking the filename in the dynamic config, we say "config lives here"
There's an initial read on start-up, and then any time a new file is moved there, Envoy will reread and (generally) reload.